mathics/builtin/assignments/internals.py

```diff
@@ -755,5 +755,21 @@ def assign(self, lhs, rhs, evaluation):
                 return False
             indices = lhs.elements[1:]
             return walk_parts([rule.replace], indices, evaluation, rhs)
+
+        # FIXME: the below is a big hack.
+        # Currently MakeBoxes boxing is implemented as a bunch of rules.
+        # See mathics.builtin.base contribute().
+        # I think we want to change this so it works like normal SetDelayed
+        # That is:
+        #   MakeBoxes[CubeRoot, StandardForm] := RadicalBox[3, StandardForm]
+        # rather than:
+        #   MakeBoxes[CubeRoot, StandardForm] -> RadicalBox[3, StandardForm]
+        elif lhs.get_head_name() == "System`MakeBoxes":
```

Comment:
If you look above, you can see that some time ago we had defined a dictionary where special kinds of LHS are associated yo different ways to perform the assignment. Maybe it would be wise to use it.

Reply:
I don't see exactly what you are talking about and this code is all temporary until the set functions are gone over. And there may be something even larger wrong in the way `contribute()` works if it works using the `rules` class variable.

That said, if you feel you have a better and cleaner way to do this, by all means go head and commit on top.

Reply:
It is just a matter of organization. And probably, you would like to reformulate it again. However, I think that the idea of using a dictionary to store the connection between different kinds of expressions and implementation was an improvement over a large spaghetti code.